src/character.h

```diff
@@ -178,6 +178,19 @@ struct social_modifiers {
         return *this;
     }
 };
+
+struct consumption_event {
+    time_point time;
+    itype_id type_id;
+    uint64_t component_hash = 0;
+
+    consumption_event( const item &food ) {
+        time = calendar::turn;
+        type_id = food.typeId();
```

Comment:
Looks like this could be included in the hash. You don't actually do anything but compare it, just like the hash.

Reply:
KorGgenT expressed interest in using this consumption history for other things (like a in-game nutrient-tracking device) later, so I figured I'd store the item type separately.